romancal/pipeline/highlevel_pipeline.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from os.path import basename
+
+from roman_datamodels import datamodels as rdm
+
+import romancal.datamodels.filetype as filetype
+
+# from romancal.resample import ResampleStep
+from romancal.datamodels import ModelContainer
+from romancal.outlier_detection import OutlierDetectionStep
+
+# step imports
+from romancal.skymatch import SkyMatchStep
+
+from ..stpipe import RomanPipeline
+
+__all__ = ["HighLevelPipeline"]
+
+# Define logging
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+class HighLevelPipeline(RomanPipeline):
+    """
+    HighLevelPipeline: Apply all calibration steps to the roman data
+    to produce level 3 products. Included steps are:
+    skymatch, Outlierdetectionn and resample.
+    """
+
+    class_alias = "roman_hlp"
+
+    spec = """
+        save_results = boolean(default=False)
+    """
+
+    # Define aliases to steps
+    step_defs = {
+        "skymatch": SkyMatchStep,
+        "outlierdet": OutlierDetectionStep,
+        # "resample": ResampleStep,
+    }
+
+    # start the actual processing
+    def process(self, input):
+        """Process the Roman WFI data from Level 2 to Level 3"""
+
+        log.info("Starting Roman high level calibration pipeline ...")
+        if isinstance(input, str):
+            input_filename = basename(input)
+        else:
+            input_filename = None
+
+        # open the input file
+        file_type = filetype.check(input)
+        asn = None
+        if file_type == "asdf":
+            try:
+                input = rdm.open(input)
+            except TypeError:
+                log.debug("Error opening file:")
+                return
+
+        if file_type == "asn":
+            asn = ModelContainer.read_asn(input)
+
+        # Build a list of observations to process
+        expos_file = []
+        if file_type == "asdf":
+            expos_file = [input]
+        elif file_type == "asn":
+            for product in asn["products"]:
+                for member in product["members"]:
+                    expos_file.append(member["expname"])
+
+        results = []
+        for in_file in expos_file:
+            if isinstance(in_file, str):
+                input_filename = basename(in_file)
+                log.info(f"Input file name: {input_filename}")
+            else:
+                input_filename = None
+
+            # Open the file
+            input = rdm.open(in_file)
+            log.info(f"Processing a WFI exposure {in_file}")
+
+            self.skymatch.suffix = "skymatch"
+            result = self.skymatch(input)
+            if input_filename:
+                result.meta.filename = input_filename
+            result = self.outlierdetection(result)
+
+            result = self.resample(result)
+            # setup output_file for saving
+            self.setup_output(result)
+            log.info("Roman exposure calibration pipeline ending...")
+
+            self.output_use_model = True
+            results.append(result)
```

Comment:
I'm confused here. I think skymatching & outlier detection need to get the full list of exposures being combined, and this looks like instead they get them one by one. I don't think skymatch & outlier detection can do anything with a single exposure, and presumably just return it back ~unchanged. I think I would have guessed that these would get sent the full asn / ModelContainer, and there wouldn't be a loop in the HLP?

Reply:
I think you are correct but at the moment I have no data for testing. So for the interim I've keep the elp behavior . My impression was that Nadia wanted this in so I went ahead and plan on fixing any issues once we have working examples. I have no problem waiting until we have input data to merge this but we'll be quite close to the release deadline.

Reply:
We have input data, the data used for testing resample. It's a FOV dithered simulation and Mairan knows the location.